### ISSUE TITLE
System test iocs: fix kepco_no_rem, smrtmon, zfctrnl

### ISSUE THREAD

**Issue by JamesKingWork** [ISISComputingGroup/IBEX#5903]
As a user, I would like the IOCs I use to be passing their tests so we have some guaranteed behaviour.

Tests failing consistently:

- [x] kepco_no_rem
- [x] smrtmon
- [x] zfctrnl

Acceptance criteria:

- [ ] These three test cases are passing locally and on build server

**Comment by JamesKingWork** [comment]
PRs:

- https://github.com/ISISComputingGroup/EPICS-kepco/pull/11
- https://github.com/ISISComputingGroup/EPICS-IOC_Test_Framework/pull/388

**Comment by kjwoodsISIS** [comment]
kepco_no_rem was fixed in #5888.  I think we need to understand why it is failing again.  Is it the same problem as before, or something different?  If it is a new problem, why didn't we spot it before?  There may be an immediate cause of the failure that we are addressing now, but repeated failures suggest to me that there is some underlying cause.  (For example, if these test worked in people's dev environments, what is causing them to fail in the test environment? Is the test environment changing in such a way as to cause test failures?  Are the tests themselves fragile and not robust against environmental changes?).

**Comment by JamesKingWork** [comment]
> kepco_no_rem was fixed in #5888. I think we need to understand why it is failing again. Is it the same problem as before, or something different? If it is a new problem, why didn't we spot it before? There may be an immediate cause of the failure that we are addressing now, but repeated failures suggest to me that there is some underlying cause. (For example, if these test worked in people's dev environments, what is causing them to fail in the test environment? Is the test environment changing in such a way as to cause test failures? Are the tests themselves fragile and not robust against environmental changes?).

After #5888 it was not failing, but then changes were made in order to allow ramping of the kepco and as such the CURRENT:SP pv was changed and the autosave field applied to the wrong record (https://github.com/ISISComputingGroup/EPICS-kepco/commit/e82121cf8a8800854eff14501fba44c178cf691b). We didn't spot it because those changes didn't make it into the build on the build machine and I didn't have those changes on my local machine at the time (I believe they weren't merged when I started looking at the failures). This is not a failure due to the test environment, just because changes were made.

**Comment by JamesKingWork** [comment]
Failure reasons:

- kepco_no_rem: CURRENT:SP was changed in https://github.com/ISISComputingGroup/EPICS-kepco/commit/e82121cf8a8800854eff14501fba44c178cf691b and autosave was applied to the wrong field
- smrtmon: Had a syntax error so not all records were loaded correctly
- zfctrnl: Relies on the kepco and changes to the CURRENT:SP caused alarms to not be set correctly

**Comment by DominicOram** [comment]
Pass locally now, I'm going to assume they pass on the build server and call this done, we can open a new ticket if they don't
